FWCore/Concurrency/interface/WaitingTask.h

```diff
@@ -57,16 +57,15 @@ namespace edm {
      * This method should only be called by WaitingTaskList
      */
     void dependentTaskFailed(std::exception_ptr iPtr) {
-      if (iPtr and not m_ptr) {
-        auto temp = std::make_unique<std::exception_ptr>(iPtr);
-        std::exception_ptr* expected = nullptr;
-        if (m_ptr.compare_exchange_strong(expected, temp.get())) {
-          temp.release();
-        }
+      bool isSet = false;
+      if (iPtr and m_ptrSet.compare_exchange_strong(isSet, true)) {
+        m_ptr = iPtr;
+        //NOTE: the atomic counter in TaskBase will synchronize m_ptr
```

Comment:
Just curious, in what kind of situation would this comment be useful?

Reply:
It explains why some sort of atomic fence isn’t needed after setting the value even though the value may be read by another thread later.